event-adapter/cmd/provider/provider.go

```diff
@@ -17,44 +17,146 @@ limitations under the License.
+	service            *sd.Service
+	maxRetrievedEvents int
+	sinceMillis        int64
 }
 
-//TODO(erocchi) add implementation to communicate with Stackdriver
-
 // NewStackdriverProvider creates a new Provider with standard settings
-func NewStackdriverProvider() provider.EventsProvider {
-	return &StackdriverProvider{}
+func NewStackdriverProvider(restClient rest.Interface, stackdriverService *sd.Service, maxRetrievedEvents int, sinceMillis int64) provider.EventsProvider {
+	return &StackdriverProvider{
+		restClient:         restClient,
+		service:            stackdriverService,
+		maxRetrievedEvents: maxRetrievedEvents,
+		sinceMillis:        sinceMillis,
+	}
 }
 
 // GetNamespacedEventsByName gets the event with the given namespace and name
 func (p *StackdriverProvider) GetNamespacedEventsByName(namespace, eventName string) (*api.Event, error) {
-	return nil, fmt.Errorf("GetNamespacedEventsByName is not implemented yet.")
+
+	standardFilter, err := standardFilter()
+	if err != nil {
+		return nil, err
+	}
+	filter := "(" + standardFilter + " AND jsonPayload.metadata.namespace = \"" + namespace + "\" AND jsonPayload.metadata.name = \"" + eventName + "\")"
+	event, err := p.doRequest(filter)
+	if err != nil {
+		return nil, err
+	}
+
+	if len(event.Items) == 0 {
+		return nil, fmt.Errorf("Event not found")
+	}
+	return &event.Items[0], nil
 }
 
 // ListAllEventsByNamespace gets all the events with the given namespace
 func (p *StackdriverProvider) ListAllEventsByNamespace(namespace string) (*api.EventList, error) {
-	return nil, fmt.Errorf("ListAllEventsByNamespace is not implemented yet.")
+
+	standardFilter, err := standardFilter()
+	if err != nil {
+		return nil, err
+	}
+
+	filter := "(" + standardFilter + " AND jsonPayload.metadata.namespace = \"" + namespace + "\")"
+	if err != nil {
+		return nil, err
+	}
+
+	return p.doRequest(filter)
 }
 
 // ListAllEvents gets all the events
 func (p *StackdriverProvider) ListAllEvents() (*api.EventList, error) {
-	return nil, fmt.Errorf("ListAllEvents is not implemented yet.")
+	filter, err := standardFilter()
+	if err != nil {
+		return nil, err
+	}
+	return p.doRequest(filter)
+}
+
+func standardFilter() (string, error) {
+	projectID, err := gce.ProjectID()
+
+	if err != nil {
+		return "", fmt.Errorf("Cannot retrieve projectID. %v", err)
+	}
+
+	return "(logName = \"projects/" + projectID + "/logs/events\" AND jsonPayload.kind = \"Event\")", nil
+
 }
 
 // CreateNewEvent creates a new event in the given namespace
 func (p *StackdriverProvider) CreateNewEvent(namespace string) (*api.Event, error) {
 	return nil, fmt.Errorf("CreateNewEvent is not implemented yet.")
 }
+
+func (p *StackdriverProvider) doRequest(filter string) (*api.EventList, error) {
+	projectID, err := gce.ProjectID()
+
+	if err != nil {
+		return nil, fmt.Errorf("Cannot retrieve projectID. %v", err)
+	}
+
+	resource := []string{
+		"projects/" + projectID,
+	}
+
+	entriesService := sd.NewEntriesService(p.service)
```

Comment:
Is there a reason why you create entriesService and newEventsTranslator each time you send a request instead of doing it a single time, for example in NewStackdriverProvider function?

Reply:
Moved